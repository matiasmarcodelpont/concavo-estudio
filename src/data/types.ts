export type WithNullProp<T, K extends keyof T> = Omit<T, K> & {
  [P in K]: null
}

export interface Reference {
  slug: string
}

export type OmitReferences<T> = {
  [K in keyof T as Exclude<T[K], null> extends Reference | Reference[] ? never : K]: T[K]
}

export interface Producto {
  name: string
  slug: string
  colaborador: Reference | null
}

export type ConcavoProducto = WithNullProp<Producto, 'colaborador'>

export function isConcavoProducto(producto: Producto): producto is ConcavoProducto {
  return producto.colaborador === null
}

export interface Ambiente {
  name: string
  slug: string
  productos: Reference[]
}

interface BaseColaborador {
  name: string
  slug: string
  website: string
}

export type MainColaborador = BaseColaborador & {
  isMain: true
  description: string
  email: string
  address: string
}

export type StandardColaborador = BaseColaborador & {
  isMain: false
  description: string | null
  email: string | null
  address: string | null
}

export type Colaborador = MainColaborador | StandardColaborador

export function isMainColaborador(colaborador: Colaborador): colaborador is MainColaborador {
  return colaborador.isMain
}

export interface DataSet {
  productos: Producto[]
  ambientes: Ambiente[]
  colaboradores: Colaborador[]
}
