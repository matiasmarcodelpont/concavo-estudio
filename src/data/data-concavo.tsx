import ConcavoColors from '@/components/domain/ConcavoColors'

const piezaDiseño = (
  <ol>
    <li>• Pieza de diseño realizada artesanalmente</li>
    <li>• Por ser un producto hecho a mano puede tener variaciones en sus medidas de +/- 2cm</li>
  </ol>
)

const garantias = (
  <article className='text-[10px]'>
    <p>
      Cuidados y garantías:
      <br />
      Los productos Cóncavo han sido fabricados con materiales de calidad superior y aseguran al consumidor una
      prolongada duración. Con el objetivo de conservar esta calidad le aconsejamos:
    </p>
    <ol>
      <li>
        - No usar productos de limpieza abrasivos como esponjas, polvos, etc. Es conveniente utilizar jabones o
        detergentes neutros
      </li>
      <li>- Los productos Cóncavo tienen una garantía de 5 años contra todo defecto de fabricación.</li>
      <li>
        - Excluye daños ocasionados por el uso inadecuado, instalaciones deficientes o el uso de limpiadores abrasivos.
      </li>
      <li>- Es requisito inexcusable para realizar cualquier reclamo, la presentación de esta garantía.</li>
    </ol>
  </article>
)

export const productosConcavo = [
  {
    slug: 'bacha-kahlo',
    name: 'Bacha Kahlo',
    description: (
      <>
        Bacha de apoyo | exenta
        <br />
        Medidas: 620 x 320 x 110 mm (h)
        <br />
        Material: concreto
        <br />
        Diámetro desagüe: 40mm
        <br />
        Color: a elección (10 colores)
        <br />
        <br />
        <ConcavoColors />
        <br />
        {piezaDiseño}
        <br />
        {garantias}
      </>
    ),
    link: null,
    colaborador: null,
  },
  {
    slug: 'bacha-testa',
    name: 'Bacha Testa',
    description: (
      <>
        Bacha de apoyo | exenta
        <br />
        Medidas: 620 x 320 x 110 mm (h)
        <br />
        Material: concreto
        <br />
        Diámetro desagüe: 40mm
        <br />
        Color: a elección (10 colores)
        <br />
        <br />
        <ConcavoColors />
        <br />
        {piezaDiseño}
        <br />
        {garantias}
      </>
    ),
    link: null,
    colaborador: null,
  },
  {
    slug: 'bacha-costa',
    name: 'Bacha Costa',
    description: (
      <>
        Bacha de apoyo | exenta
        <br />
        Medidas: diametro 360 x 110 mm (h)
        <br />
        Material: concreto
        <br />
        Diámetro desagüe: 40mm
        <br />
        Color: a elección (10 colores)
        <br />
        <br />
        <ConcavoColors />
        <br />
        {piezaDiseño}
        <br />
        {garantias}
      </>
    ),
    link: null,
    colaborador: null,
  },
  {
    slug: 'bacha-dieste',
    name: 'Bacha Dieste',
    description: (
      <>
        Bacha de apoyo | exenta
        <br />
        Medidas: diametro 360 x 60 mm (h)
        <br />
        Material: concreto
        <br />
        Diámetro desagüe: 40mm
        <br />
        Color: a elección (10 colores)
        <br />
        <br />
        <ConcavoColors />
        <br />
        {piezaDiseño}
        <br />
        {garantias}
      </>
    ),
    link: null,
    colaborador: null,
  },
  {
    slug: 'bañadera-kahlo',
    name: 'Bañadera Kahlo',
    description: (
      <>
        Bacha de apoyo | exenta
        <br />
        Medidas: 1750 x 860 x 470 mm (h)
        <br />
        Material: concreto
        <br />
        Diámetro desagüe: 52mm
        <br />
        Color: a elección (10 colores)
        <br />
        <br />
        <ConcavoColors />
        <br />
        {piezaDiseño}
        <br />
        {garantias}
      </>
    ),
    link: null,
    colaborador: null,
  },
  {
    slug: 'bañadera-testa',
    name: 'Bañadera Testa',
    description: (
      <>
        Bacha de apoyo | exenta
        <br />
        Medidas: 1570 x 780 x 440 mm (h)
        <br />
        Material: concreto
        <br />
        Diámetro desagüe: 52mm
        <br />
        Color: a elección (10 colores)
        <br />
        <br />
        <ConcavoColors />
        <br />
        {piezaDiseño}
        <br />
        {garantias}
      </>
    ),
    link: null,
    colaborador: null,
  },
  {
    slug: 'lampara-dieste',
    name: 'Lámpara Dieste',
    description: (
      <>
        <strong>Podes elegir opción colgante de techo o aplique de pared y combinarlas como más te guste</strong>
        <br />
        <br />
        <strong>(L) Large</strong>
        <br />
        Medidas: diámetro 360 x 60 mm (h)
        <br />
        <br />
        <strong>(S) Small</strong>
        <br />
        Medidas: diámetro 160 x 60 mm (h)
        <br />
        <br />
        Material: Concreto
        <br />
        Color: a elección (10 colores)
        <br />
        <br />
        <ConcavoColors />
        <br />
        Foco rosca tipo E27 Recomendamos luz cálida (lámpara no incluida)
        <br />
        Incluye Florón para salida de cable en techo de 10cm de diámetro
        <br />
        Opciones: Cable blanco o negro
        <br />
        <br />
        {piezaDiseño}
        <br />
        {garantias}
      </>
    ),
    link: null,
    colaborador: null,
  },
  {
    slug: 'aplique-botero',
    name: 'Aplique Botero',
    description: (
      <>
        <br />
        Aplique de pared
        <br />
        Medidas: 250 x 130 x 60 mm (profundidad)
        <br />
        Material: Concreto
        <br />
        Color: a elección (10 colores)
        <br />
        <br />
        <ConcavoColors />
        <br />
        Foco tipo bipin G9 220v
        <br />
        Recomendamos luz cálida (lámpara no incluida)
        <br />
        <br />
        {piezaDiseño}
        <br />
        {garantias}
      </>
    ),
    link: null,
    colaborador: null,
  },
]
