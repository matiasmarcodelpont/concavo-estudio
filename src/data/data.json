{
  "rooms": [
    {
      "slug": "living-comedor",
      "name": "Living/Comedor",
      "products": [
        {
          "slug": "lampara-hierro"
        },
        {
          "slug": "piso-madera"
        }
      ],
      "contributors": []
    },
    {
      "slug": "cocina",
      "name": "Cocina",
      "products": [
        {
          "slug": "azulejos-guardados"
        }
      ],
      "contributors": []
    },
    {
      "slug": "lavadero",
      "name": "Lavadero",
      "products": [],
      "contributors": []
    },
    {
      "slug": "escritorio",
      "name": "Escritorio",
      "products": [
        {
          "slug": "lampara-hierro"
        },
        {
          "slug": "piso-madera"
        }
      ],
      "contributors": []
    },
    {
      "slug": "toilet",
      "name": "Toilet",
      "products": [],
      "contributors": []
    },
    {
      "slug": "cuarto-alegra",
      "name": "Cuarto Alegra",
      "products": [],
      "contributors": []
    },
    {
      "slug": "baño-alegra",
      "name": "Baño Alegra",
      "products": [],
      "contributors": []
    },
    {
      "slug": "cuarto-principal",
      "name": "Cuarto principal",
      "products": [],
      "contributors": []
    },
    {
      "slug": "baño-principal",
      "name": "Baño principal",
      "products": [],
      "contributors": []
    },
    {
      "slug": "galeria",
      "name": "Galería",
      "products": [],
      "contributors": []
    },
    {
      "slug": "jardin",
      "name": "Jardín",
      "products": [],
      "contributors": []
    }
  ],
  "products": [
    {
      "slug": "piso-madera",
      "name": "Piso simil Madera",
      "contributor": {
        "slug": "elegance"
      }
    },
    {
      "slug": "lampara-hierro",
      "name": "Lámpara de diseño con detalles de Hierro",
      "contributor": {
        "slug": "luz-viva"
      }
    },
    {
      "slug": "sofa-carlos",
      "name": "Sofá Carlos V",
      "contributor": {
        "slug": "muebles-arte"
      }
    },
    {
      "slug": "azulejos-guardados",
      "name": "Azulejos Guardados",
      "contributor": {
        "slug": "cocina-design"
      }
    },
    {
      "slug": "bacha-8080",
      "name": "Bacha 8080",
      "contributor": null
    },
    {
      "slug": "bañadera-10001",
      "name": "Bañadera 10001",
      "contributor": null
    }
  ],
  "contributors": [
    {
      "slug": "elegance",
      "name": "Elegance",
      "website": "www.pisoselegance.es",
      "isMain": true,
      "description": "Líder en revestimientos de suelos elegantes y duraderos para hogares y negocios. Calidad, innovación y compromiso con la sostenibilidad nos distinguen.",
      "email": "info@pisoselegance.com",
      "address": "Avenida del Diseño 123, Ciudad Elegante, España"
    },
    {
      "slug": "luz-viva",
      "name": "Luz Viva",
      "website": "www.luzviva.com",
      "isMain": true,
      "description": "Líder en sistemas de iluminación innovadores y eficientes para hogares y negocios. Calidad, diseño y compromiso con la sostenibilidad nos distinguen.",
      "email": "info@luzviva.com",
      "address": "Calle de la Luz 456, Ciudad Brillante, México"
    },
    {
      "slug": "muebles-arte",
      "name": "Muebles de Arte",
      "website": "www.mueblesdearte.com",
      "isMain": false,
      "description": null,
      "email": null,
      "address": "Calle del Diseño 789, Ciudad Creativa, Argentina"
    },
    {
      "slug": "cocina-design",
      "name": "Cocina Design",
      "website": "www.cocinadesign.com",
      "isMain": false,
      "description": "Especialistas en diseño y equipamiento de cocinas de alta gama. Ofrecemos soluciones personalizadas para crear la cocina de tus sueños, combinando estilo, funcionalidad y calidad.",
      "email": null,
      "address": null
    }
  ]
}
