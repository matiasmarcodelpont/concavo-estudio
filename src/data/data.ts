import { DataSet } from './types'

const data: DataSet = {
  ambientes: [
    {
      slug: 'living-comedor',
      name: 'Living/Comedor',
      productos: [
        {
          slug: 'lampara-hierro',
        },
        {
          slug: 'piso-madera',
        },
      ],
      images: [
        {
          src: '0.jpeg',
          width: 1500,
          height: 1080,
          puntitos: [{ productoSlug: 'azulejos-guardados', coordinates: { x: 760, y: 430 } }],
        },
        { src: '1.jpeg', width: 1500, height: 2000, puntitos: [] },
        { src: '2.jpeg', width: 1500, height: 1080, puntitos: [] },
      ],
    },
    {
      slug: 'cocina',
      name: 'Cocina',
      productos: [
        {
          slug: 'azulejos-guardados',
        },
      ],
      images: [
        { src: '0.jpeg', width: 1500, height: 1080, puntitos: [] },
        { src: '1.jpeg', width: 1500, height: 2000, puntitos: [] },
        { src: '2.jpeg', width: 1500, height: 1080, puntitos: [] },
      ],
    },
    {
      slug: 'lavadero',
      name: 'Lavadero',
      productos: [],
      images: [
        { src: '0.jpeg', width: 1500, height: 2000, puntitos: [] },
        { src: '1.jpeg', width: 1500, height: 1080, puntitos: [] },
        { src: '2.jpeg', width: 1500, height: 1080, puntitos: [] },
      ],
    },
    // {
    //   slug: 'escritorio',
    //   name: 'Escritorio',
    //   productos: [
    //     {
    //       slug: 'lampara-hierro',
    //     },
    //     {
    //       slug: 'piso-madera',
    //     },
    //   ],
    // },
    // {
    //   slug: 'toilet',
    //   name: 'Toilet',
    //   productos: [],
    // },
    // {
    //   slug: 'cuarto-alegra',
    //   name: 'Cuarto Alegra',
    //   productos: [],
    // },
    // {
    //   slug: 'baño-alegra',
    //   name: 'Baño Alegra',
    //   productos: [],
    // },
    // {
    //   slug: 'cuarto-principal',
    //   name: 'Cuarto principal',
    //   productos: [],
    // },
    // {
    //   slug: 'baño-principal',
    //   name: 'Baño principal',
    //   productos: [],
    // },
    // {
    //   slug: 'galeria',
    //   name: 'Galería',
    //   productos: [],
    // },
    // {
    //   slug: 'jardin',
    //   name: 'Jardín',
    //   productos: [],
    // },
  ],
  productos: [
    {
      slug: 'piso-madera',
      name: 'Piso simil Madera',
      description:
        'Lorem ipsum dolor sit amet consectetur adipisicing elit. Incidunt voluptas explicabo aliquam tenetur. Eligendi deleniti ratione doloribus dolores dolorum, reprehenderit maiores odit itaque quos tenetur, architecto earum est, iste accusantium.',
      link: 'https://example.com',
      colaborador: {
        slug: 'elegance',
      },
    },
    {
      slug: 'lampara-hierro',
      name: 'Lámpara de Hierro',
      description:
        'Lorem ipsum dolor sit amet consectetur adipisicing elit. Incidunt voluptas explicabo aliquam tenetur. Eligendi deleniti ratione doloribus dolores dolorum, reprehenderit maiores odit itaque quos tenetur, architecto earum est, iste accusantium.',
      link: 'https://example.com',
      colaborador: {
        slug: 'luz-viva',
      },
    },
    {
      slug: 'sofa-carlos',
      name: 'Sofá Carlos V',
      description:
        'Lorem ipsum dolor sit amet consectetur adipisicing elit. Incidunt voluptas explicabo aliquam tenetur. Eligendi deleniti ratione doloribus dolores dolorum, reprehenderit maiores odit itaque quos tenetur, architecto earum est, iste accusantium.',
      link: 'https://example.com',
      colaborador: {
        slug: 'muebles-arte',
      },
    },
    {
      slug: 'azulejos-guardados',
      name: 'Azulejos Guardados',
      link: 'https://example.com',
      description:
        'Lorem ipsum dolor sit amet consectetur adipisicing elit. Incidunt voluptas explicabo aliquam tenetur. Eligendi deleniti ratione doloribus dolores dolorum, reprehenderit maiores odit itaque quos tenetur, architecto earum est, iste accusantium.',

      colaborador: {
        slug: 'cocina-design',
      },
    },
    {
      slug: 'bacha-costa',
      name: 'Bacha Costa',
      link: 'https://example.com',
      description:
        'Lorem ipsum dolor sit amet consectetur adipisicing elit. Incidunt voluptas explicabo aliquam tenetur. Eligendi deleniti ratione doloribus dolores dolorum, reprehenderit maiores odit itaque quos tenetur, architecto earum est, iste accusantium.',
      colaborador: null,
    },
    {
      slug: 'bañadera-kahlo',
      name: 'Bañadera Kahlo',
      link: 'https://example.com',
      description:
        'Lorem ipsum dolor sit amet consectetur adipisicing elit. Incidunt voluptas explicabo aliquam tenetur. Eligendi deleniti ratione doloribus dolores dolorum, reprehenderit maiores odit itaque quos tenetur, architecto earum est, iste accusantium.',
      colaborador: null,
    },
    {
      slug: 'bañadera-kahlo',
      name: 'Bañadera Kahlo',
      link: 'https://example.com',
      description:
        'Lorem ipsum dolor sit amet consectetur adipisicing elit. Incidunt voluptas explicabo aliquam tenetur. Eligendi deleniti ratione doloribus dolores dolorum, reprehenderit maiores odit itaque quos tenetur, architecto earum est, iste accusantium.',
      colaborador: null,
    },
    {
      slug: 'bañadera-kahlo',
      name: 'Bañadera Kahlo',
      link: 'https://example.com',
      description:
        'Lorem ipsum dolor sit amet consectetur adipisicing elit. Incidunt voluptas explicabo aliquam tenetur. Eligendi deleniti ratione doloribus dolores dolorum, reprehenderit maiores odit itaque quos tenetur, architecto earum est, iste accusantium.',
      colaborador: null,
    },
  ],
  colaboradores: [
    {
      slug: 'elegance',
      name: 'Elegance',
      contact: 'Contacto: 5473-1862',
    },
    {
      slug: 'luz-viva',
      name: 'Luz Viva',
      contact: 'Contacto: 5473-1862',
    },
    {
      slug: 'muebles-arte',
      name: 'Muebles de Arte',
      contact: 'Contacto: 5473-1862',
    },
    {
      slug: 'cocina-design',
      name: 'Cocina Design',
      contact: 'Contacto: 5473-1862',
    },
  ],
}

export default data
