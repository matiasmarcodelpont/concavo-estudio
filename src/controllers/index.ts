import data from '@/data/data.json'

import { DataSet } from '@/data/types'

import { createProductsRepository } from '@/data/repositories/products'
import { createRoomsRepository } from '@/data/repositories/rooms'
import { createContributorsRepository } from '@/data/repositories/contributors'

export const productsRepository = createProductsRepository(data as DataSet)

export const roomsRepository = createRoomsRepository(data as DataSet)

export const contributorsRepository = createContributorsRepository(data as DataSet)
