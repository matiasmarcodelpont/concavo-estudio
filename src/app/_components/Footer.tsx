'use client'

import { Qr } from '@/components/ui/qr'
import { ambientesRepository, productosRepository } from '@/controllers'
import { useHandleQr } from './hooks'
import LogoAndSocialMedia from '@/components/domain/LogoAndSocialMedia'

export const Footer = () => {
  const ambientes = ambientesRepository.getAmbientes()
  const productos = productosRepository.getProductosConcavo()

  const { currentUrl, handleClick, qrRef } = useHandleQr()

  const siteMap = [
    {
      name: 'Home',
      links: [
        {
          name: 'Cóncavo',
          slug: 'concavo',
          href: '/',
        },
        {
          name: 'Productos Cóncavo',
          slug: 'productos-concavo',
          href: '/#productos-concavo',
        },
        {
          name: 'Casa Cóncavo',
          slug: 'casa-concavo',
          href: '/casa-concavo',
        },
      ],
    },
    {
      name: 'Casa Concavo',
      links: ambientes.map(({ name, slug }) => ({ name, slug, href: `/casa-concavo/${slug}` })),
    },
    {
      name: 'Productos',
      links: productos.map(({ name, slug }) => ({ name, slug, href: `/productos/${slug}` })),
    },
  ]

  return (
    <footer className='bg-white text-black flex gap-5 md:gap-10 px-6 sm:px-10 md:px-12 pt-10 pb-6 flex-wrap-reverse'>
      {siteMap.map(({ name, links }, index) => (
        <section key={`sitemap-${index.toString()}-${name}`} className='hidden md:block'>
          <h1 className='text-base lg:text-xl font-tt-norms font-bold uppercase mb-2'>{name}</h1>
          <ul className='text-xs gap-y-2 gap-x-6 flex flex-col flex-wrap max-h-[150px]'>
            {links.map(({ name, slug, href }) => (
              <li key={slug} className='hover:underline'>
                <a href={href}>{name}</a>
              </li>
            ))}
          </ul>
        </section>
      ))}

      <section className='flex gap-6 xs:gap-10 sm:gap-14 ml-auto mr-auto md:mr-0 flex-wrap'>
        <LogoAndSocialMedia variant='black' className='mx-auto' />
        <div className='mx-auto'>
          <Qr value={currentUrl} onClick={handleClick} />
        </div>
        <div className='absolute left-[-100vw] top-0'>
          <Qr ref={qrRef} value={currentUrl} size={300} />
        </div>
      </section>
    </footer>
  )
}
