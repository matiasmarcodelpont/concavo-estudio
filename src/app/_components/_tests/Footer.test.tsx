import { render } from '@testing-library/react'

import { Footer } from '../Footer'
import { productosRepository } from '@/controllers'
import { Producto } from '@/data/types'

jest.mock('@/controllers', () => ({
  productosRepository: {
    getProductosInAmbiente: jest.fn(),
  },
}))

describe('Footer', () => {
  type GetProductosInAmbiente = (slug: string) => Producto[]
  ;(productosRepository.getProductosInAmbiente as jest.MockedFunction<GetProductosInAmbiente>).mockImplementation(
    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
    // @ts-expect-error
    (slug) => {
      return [
        { name: `Producto 1 in ${slug}`, slug: `producto-1-${slug}` },
        { name: `Producto 2 in ${slug}`, slug: `producto-2-${slug}` },
      ]
    },
  )

  const ambientes = [
    { name: 'Ambiente 1', slug: 'ambiente-1' },
    { name: 'Ambiente 2', slug: 'ambiente-2' },
  ]

  it('matches snapshot', () => {
    const { getByRole } = render(<Footer ambientes={ambientes} />)
    const component = getByRole('contentinfo')

    expect(component).toMatchSnapshot()
  })

  it('has the home links', () => {
    const { getByRole } = render(<Footer ambientes={ambientes} />)

    const anchor1 = getByRole('link', { name: 'C贸ncavo' }) as HTMLAnchorElement
    expect(anchor1.href).toMatch(/^https?:\/\/[^/]+\/$/)

    const anchor2 = getByRole('link', { name: 'Productos C贸ncavo' }) as HTMLAnchorElement
    expect(anchor2.href).toMatch(/^https?:\/\/[^/]+\/#productos-concavo$/)

    const anchor3 = getByRole('link', { name: 'Casa C贸ncavo' }) as HTMLAnchorElement
    expect(anchor3.href).toMatch(/^https?:\/\/[^/]+\/casa$/)
  })

  it('has the ambiente links', () => {
    const { getByRole } = render(<Footer ambientes={ambientes} />)

    const anchor1 = getByRole('link', { name: 'Ambiente 1' }) as HTMLAnchorElement
    expect(anchor1.href).toMatch(/^https?:\/\/[^/]+\/casa\/ambiente-1$/)

    const anchor2 = getByRole('link', { name: 'Ambiente 2' }) as HTMLAnchorElement
    expect(anchor2.href).toMatch(/^https?:\/\/[^/]+\/casa\/ambiente-2$/)
  })

  it('has the producto links', () => {
    const { getByRole } = render(<Footer ambientes={ambientes} />)

    const anchor1 = getByRole('link', { name: 'Producto 1 in ambiente-1' }) as HTMLAnchorElement
    expect(anchor1.href).toMatch(/^https?:\/\/[^/]+\/productos\/producto-1-ambiente-1$/)

    const anchor2 = getByRole('link', { name: 'Producto 2 in ambiente-1' }) as HTMLAnchorElement
    expect(anchor2.href).toMatch(/^https?:\/\/[^/]+\/productos\/producto-2-ambiente-1$/)

    const anchor3 = getByRole('link', { name: 'Producto 1 in ambiente-2' }) as HTMLAnchorElement
    expect(anchor3.href).toMatch(/^https?:\/\/[^/]+\/productos\/producto-1-ambiente-2$/)

    const anchor4 = getByRole('link', { name: 'Producto 2 in ambiente-2' }) as HTMLAnchorElement
    expect(anchor4.href).toMatch(/^https?:\/\/[^/]+\/productos\/producto-2-ambiente-2$/)
  })

  it('has Concavo logo', () => {
    const { getByAltText } = render(<Footer ambientes={ambientes} />)
    getByAltText('Logo C贸ncavo')
  })

  it('has social media links', () => {
    const { getByLabelText } = render(<Footer ambientes={ambientes} />)

    const instagramAnchor = getByLabelText('instagram') as HTMLAnchorElement
    expect(instagramAnchor.href).toBe('https://www.instagram.com/concavoestudio/')

    const mailAmchor = getByLabelText('email') as HTMLAnchorElement
    expect(mailAmchor.href).toBe('mailto:ventas@concavoestudio.com')
  })
})
