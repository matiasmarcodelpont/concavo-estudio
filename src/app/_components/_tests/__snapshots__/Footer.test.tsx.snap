// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer matches snapshot 1`] = `
<footer
  class="bg-white text-black flex gap-10 sm:gap-14 px-6 sm:px-10 md:px-12 pt-10 pb-6 flex-wrap"
>
  <section>
    <h1
      class="font-tt-norms font-bold uppercase mb-2"
    >
      Home
    </h1>
    <ul
      class="text-xs gap-y-2 gap-x-6 flex flex-col flex-wrap max-h-[150px]"
    >
      <li
        class="hover:underline"
      >
        <a
          href="/"
        >
          C贸ncavo
        </a>
      </li>
      <li
        class="hover:underline"
      >
        <a
          href="/#productos-concavo"
        >
          Productos C贸ncavo
        </a>
      </li>
      <li
        class="hover:underline"
      >
        <a
          href="/casa-concavo"
        >
          Casa C贸ncavo
        </a>
      </li>
    </ul>
  </section>
  <section>
    <h1
      class="font-tt-norms font-bold uppercase mb-2"
    >
      Casa Concavo
    </h1>
    <ul
      class="text-xs gap-y-2 gap-x-6 flex flex-col flex-wrap max-h-[150px]"
    >
      <li
        class="hover:underline"
      >
        <a
          href="/casa-concavo/ambiente-1"
        >
          Ambiente 1
        </a>
      </li>
      <li
        class="hover:underline"
      >
        <a
          href="/casa-concavo/ambiente-2"
        >
          Ambiente 2
        </a>
      </li>
    </ul>
  </section>
  <section>
    <h1
      class="font-tt-norms font-bold uppercase mb-2"
    >
      Productos Concavo
    </h1>
    <ul
      class="text-xs gap-y-2 gap-x-6 flex flex-col flex-wrap max-h-[150px]"
    >
      <li
        class="hover:underline"
      >
        <a
          href="/productos/producto-1"
        >
          Producto 1
        </a>
      </li>
      <li
        class="hover:underline"
      >
        <a
          href="/productos/producto-2"
        >
          Producto 2
        </a>
      </li>
    </ul>
  </section>
  <section
    class="flex gap-6 sm:gap-10 ml-auto"
  >
    <div
      class="flex flex-col items-center gap-2"
    >
      <div
        class="relative w-[150px] h-[50px] sm:w-[150px] sm:h-[50px]"
      >
        <img
          alt="Logo C贸ncavo"
          class="object-contain"
          data-nimg="fill"
          decoding="async"
          loading="lazy"
          sizes="100vw"
          src="/_next/image?url=%2Flogos%2Fconcavo_black.png&w=3840&q=75"
          srcset="/_next/image?url=%2Flogos%2Fconcavo_black.png&w=640&q=75 640w, /_next/image?url=%2Flogos%2Fconcavo_black.png&w=750&q=75 750w, /_next/image?url=%2Flogos%2Fconcavo_black.png&w=828&q=75 828w, /_next/image?url=%2Flogos%2Fconcavo_black.png&w=1080&q=75 1080w, /_next/image?url=%2Flogos%2Fconcavo_black.png&w=1200&q=75 1200w, /_next/image?url=%2Flogos%2Fconcavo_black.png&w=1920&q=75 1920w, /_next/image?url=%2Flogos%2Fconcavo_black.png&w=2048&q=75 2048w, /_next/image?url=%2Flogos%2Fconcavo_black.png&w=3840&q=75 3840w"
          style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
        />
      </div>
      <div
        class="flex gap-2"
      >
        <a
          aria-label="instagram"
          class="rounded-full transition-all border-[1px] p-2 border-bone hover:border-black"
          href="https://www.instagram.com/concavoestudio/"
          target="_blank"
        >
          <svg
            class="lucide lucide-instagram "
            fill="none"
            height="20"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <rect
              height="20"
              rx="5"
              ry="5"
              width="20"
              x="2"
              y="2"
            />
            <path
              d="M16 11.37A4 4 0 1 1 12.63 8 4 4 0 0 1 16 11.37z"
            />
            <line
              x1="17.5"
              x2="17.51"
              y1="6.5"
              y2="6.5"
            />
          </svg>
        </a>
        <a
          aria-label="email"
          class="rounded-full transition-all border-[1px] p-2 border-bone hover:border-black"
          href="mailto:ventas@concavoestudio.com"
          target="_blank"
        >
          <svg
            class="lucide lucide-mail "
            fill="none"
            height="20"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <rect
              height="16"
              rx="2"
              width="20"
              x="2"
              y="4"
            />
            <path
              d="m22 7-8.97 5.7a1.94 1.94 0 0 1-2.06 0L2 7"
            />
          </svg>
        </a>
      </div>
    </div>
    <div
      data-testid="mock-qr"
    >
      Mock QR Code: 
      null
    </div>
    <div
      class="absolute left-[-100vw]"
    >
      <div
        data-testid="mock-qr"
      >
        Mock QR Code: 
        null
      </div>
    </div>
  </section>
</footer>
`;
